pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -180,13 +180,16 @@ func determineSeed(shoot *gardencorev1beta1.Shoot, seedLister gardencorelisters.
 
 func determineBestSeedCandidate(shoot *gardencorev1beta1.Shoot, cloudProfile *gardencorev1beta1.CloudProfile, shootList []*gardencorev1beta1.Shoot, seedList []*gardencorev1beta1.Seed, strategy config.CandidateDeterminationStrategy) (*gardencorev1beta1.Seed, error) {
 	var candidates []*gardencorev1beta1.Seed
-	switch strategy {
-	case config.SameRegion:
+
+	switch {
+	case shoot.Spec.Purpose != nil && *shoot.Spec.Purpose == gardencorev1beta1.ShootPurposeTesting:
```

Comment:
Lets introduce new scheduling strategy `MinimalDistanceWithLeastUsedSeedForTestingPurposeShoots`. It will choose the least loaded seed for shoots with testing purpose and minimal distance for the other shoots. 
```golang
switch strategy {
...
case config.MinimalDistanceWithLeastUsedSeedForTestingPurposeShoots:
        if shoot.Spec.Purpose != nil && *shoot.Spec.Purpose == gardencorev1beta1.ShootPurposeTesting {
            candidates = determineCandidatesOfSameProvider(seedList, shoot, candidates)
        } else {
            candidates = determineCandidatesWithSameRegionStrategy(seedList, shoot, candidates)
        }
```

@danielfoehrKn @timuthy @ialidzhikov any feedback about introducing new scheduling strategy vs. hard-coded scheduling of testing purpose? 
My point is that via new strategy Gardener operator can easily switch between the old and new behavior.

Reply:
I think a scheduling strategy makes sense in general but `MinimalDistanceWithLeastUsedSeedForTestingPurposeShoots` as an aggregated strategy sounds difficult to maintain, especially if we add more strategies in the future. So if we decided to implement this as a strategy we should make the scheduler capable of handling multiple in a configured order. But we can also do this later.

Reply:
I think kube-scheduler divides the scheduling in 2 operation - Filtering and Scoring.
Generally speaking as a potential future improvement - we could have configurable Filtering plugins that respect Seeds which are invisible/protected and remove them from the list of potential Seeds. Then we could have multiple Scoring plugins that give scores to the seed candidates. `LeastUsedSeedForTestingPurposeShoots` would be a scoring plugin and when enabled would add to the score of the least used Seed.

Reply:
cc @danielfoehrKn as he wanted to look into scheduler refactorings of this kind after he implemented the gardenlet cert rotation